[Nereus Ng Wei Bin] iP

NereusWB922 opened the pull request:
## Corgi Task Management Chatbot

> "Simplify your task management with Corgi, the chatbot that's got your back!"

Corgi is a versatile chatbot designed for efficient task management. It supports various task types, including:

- To-Do
- Deadline
- Event

### Getting Started

To use Corgi, follow these simple steps:

1. **Download Corgi**: You can get the latest version by downloading the [Corgi Jar file](download_link).

2. **Run Corgi**: Execute `java -jar corgi.jar` in your terminal or command prompt to start Corgi:

### Key Features

Corgi offers a range of features to help you stay organized:

- Task creation and management
- Deadline tracking
- Event scheduling

If you're a Java programmer, you can even extend Corgi's functionality. Here's the main Java method:
 ```java
    public static void main(String[] args) {
        Corgi bot = new Corgi();
        bot.start();
    }
```

### Todo

- [x] ~~Task Management by Corgi~~
- [ ] Teach Corgi to predict future 🔮
- [ ]  Train Corgi to make coffee ☕

:rocket: **Feel free to give Corgi a try and let us know how it can simplify your task management!** :memo:

> "Don't just manage tasks, conquer them." – Corgi

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

NereusWB922 committed:
rename, greet, exit

NereusWB922 committed:
echo

NereusWB922 committed:
Add tasks list field

NereusWB922 committed:
Add addTask method

NereusWB922 committed:
Add displayTask method

NereusWB922 committed:
Add JavaDoc for start, addTask and displayTask

NereusWB922 committed:
Add JavaDoc for constructor

NereusWB922 committed:
Add Task class

NereusWB922 committed:
Create package for tasks

NereusWB922 committed:
Ignore .class file

NereusWB922 committed:
Use Task class to represent class

NereusWB922 committed:
Add mark task done command

NereusWB922 committed:
Add mark task as not done method

NereusWB922 committed:
Add JavaDoc for markTaskAsDone , markTaskAsNotDone

NereusWB922 committed:
Add ToDo class

NereusWB922 committed:
Add Deadline class

NereusWB922 committed:
Add Event class

NereusWB922 committed:
Add addTask with different task type

NereusWB922 committed:
Add JavaDoc for overloaded addTask method

NereusWB922 committed:
Add JavaDoc to Task class

NereusWB922 committed:
Add JavaDoc to ToDo class

NereusWB922 committed:
Add JavaDoc to Deadline class

NereusWB922 committed:
Add JavaDoc to Event class

NereusWB922 committed:
Edit unit test script

NereusWB922 committed:
Add test input and expected output

NereusWB922 committed:
Add CommandType enum

NereusWB922 committed:
Add InvalidCommandException

NereusWB922 committed:
Handle invalid command error

NereusWB922 committed:
Add JavaDoc for printException method

NereusWB922 committed:
Add JavaDoc for CommandType enum

NereusWB922 committed:
Add JavaDoc for InvalidCommandException class

NereusWB922 committed:
Handle invalid command for bye and list

NereusWB922 committed:
Add parent exception class for command exception

NereusWB922 committed:
Add empty decription exception

NereusWB922 committed:
Add invalid description format exception

NereusWB922 committed:
Add command format field

NereusWB922 committed:
Handle invalid command format exception

NereusWB922 committed:
Add angry-dog themed reply

NereusWB922 committed:
Handle general exception by output to console

NereusWB922 committed:
Fix typo

NereusWB922 committed:
Print more information about general exception

NereusWB922 committed:
Edit error message

NereusWB922 committed:
Handle string to int parsing exception

NereusWB922 committed:
Add TaskStatusException class

NereusWB922 committed:
Handle TaskStatusException in mark and unmark

NereusWB922 committed:
Edit error message

NereusWB922 committed:
Edit output message

NereusWB922 committed:
Edit unit test script

NereusWB922 committed:
Edit ouptut layout

NereusWB922 committed:
Add DELETE CommandType

NereusWB922 committed:
Add delete task method

NereusWB922 committed:
Add delete command

NereusWB922 committed:
Edit remove task message

NereusWB922 committed:
Add Storable interface

NereusWB922 committed:
Make Task class abstract

NereusWB922 committed:
Add argument for parseStorableString method

NereusWB922 committed:
Remove parse method

NereusWB922 committed:
Implement Storable interface

NereusWB922 committed:
Add abstract Parser class

NereusWB922 committed:
Add custom exception for general parsing error

NereusWB922 committed:
Add InvalidParsingFormatException

NereusWB922 committed:
Add InvalidParsingTypeException

NereusWB922 committed:
Add parser for Task class

NereusWB922 committed:
Fix task type typo

NereusWB922 committed:
Handling parsing exception

NereusWB922 committed:
Add Storage class

NereusWB922 committed:
Use storage in chatbot

NereusWB922 committed:
Merge level-7 branch to master

NereusWB922 committed:
Remove never used imported class

NereusWB922 committed:
Edit format for command

NereusWB922 committed:
Add constant DateTimeFormatter

NereusWB922 committed:
Store date in LocalDate type

NereusWB922 committed:
Parse date string to LocalDate

NereusWB922 committed:
Parse date string to LocalDate type

NereusWB922 committed:
Edit console message

NereusWB922 committed:
Print message indicate new data file created

NereusWB922 committed:
Add new command to find tasks on date

NereusWB922 committed:
Add isHappeningOnDate method

NereusWB922 committed:
Add new command DATE

NereusWB922 committed:
Merge Level-8 branch to master

NereusWB922 committed:
Copy list value using constructor

NereusWB922 committed:
Add StorableList interface

NereusWB922 committed:
Implement TaskList class

NereusWB922 committed:
Create custom TaskListIndexOutOfBoundsException

NereusWB922 committed:
Modify save method argument to StorableList type

NereusWB922 committed:
Use TaskList class to replace List

NereusWB922 committed:
Replace printTask with toString

NereusWB922 committed:
Enable create exception with custom message

NereusWB922 committed:
Add CommandFormat class to store command format

NereusWB922 committed:
Add ErrorMessage class to store error message

NereusWB922 committed:
Remove \n after last task

NereusWB922 committed:
Add Ui class to extract user interface handling

NereusWB922 committed:
Use Ui class to replace print statement

NereusWB922 committed:
Add general task exception class

NereusWB922 committed:
Inherit from TaskException

NereusWB922 committed:
Add abstract Command class to represent command

NereusWB922 committed:
Implement AddTaskCommand class

NereusWB922 committed:
Implement DeleteTaskCommand class

NereusWB922 committed:
Implement ExitCommand class

NereusWB922 committed:
Implement FindTasksOnDateCommand class

NereusWB922 committed:
Implement ListTasksCommand class

NereusWB922 committed:
Implement MarkTaskCommand class

NereusWB922 committed:
Add custom exception for command execution

NereusWB922 committed:
Implement CommandParser class

NereusWB922 committed:
Add ParsingException subclass

NereusWB922 committed:
Change ParsingException to checked exception

NereusWB922 committed:
Add throws statement in parse method

NereusWB922 committed:
Edit error message correspond to exception

NereusWB922 committed:
Edit ui display

NereusWB922 committed:
Use Command class and CommandParser class

NereusWB922 committed:
Merge branch-A-MoreOOP into master

NereusWB922 committed:
Add all packages under corgi folder

NereusWB922 committed:
Add gradle to master

NereusWB922 committed:
Config build.gradle main class

NereusWB922 committed:
Add unit test for Task class

NereusWB922 committed:
Add unit test for commandParser

NereusWB922 committed:
Merge A-JUnit into master

NereusWB922 committed:
Edit shadow base name

NereusWB922 committed:
Add javadoc

NereusWB922 committed:
Remove commandFormat mapper

NereusWB922 committed:
Add javadoc

NereusWB922 committed:
Follow coding standard

NereusWB922 committed:
Extract predicate logic to FindTasksOnDateCommand

NereusWB922 committed:
Add new command type find

NereusWB922 committed:
Implement new command class for find command

NereusWB922 committed:
Edit javadoc

NereusWB922 committed:
Include logic for parsing find command

NereusWB922 committed:
Add method contains

NereusWB922 committed:
Add ui method for displaying find command result

NereusWB922 committed:
Edit import statement arrangement

NereusWB922 committed:
Merge branch-A-JavaDoc to master

NereusWB922 committed:
Merge branch-A-CodingStandard into master

NereusWB922 committed:
Merge branch-Level-9 into master

NereusWB922 committed:
Rename class

NereusWB922 committed:
Add checkstyle config file

NereusWB922 committed:
Ignore vscode file review control

NereusWB922 committed:
Setup gradle for checkstyle

NereusWB922 committed:
Merge bracnh-A-CheckStyle into master

NereusWB922 committed:
Fix checkstyle error

NereusWB922 committed:
Setup Java FX

ketweeen commented on src/main/java/corgi/Corgi.java:
```diff
@@ -0,0 +1,84 @@
+            }
+
+            this.ui.showEndLine();
+        }
```
Nice happy path!

ketweeen commented on src/main/java/corgi/commands/CommandType.java:
```diff
@@ -0,0 +1,72 @@
+/**
+ * Types of commands that can be given to the Corgi chat bot.
+ */
+public enum CommandType {
```
Nice naming for enum!

ketweeen commented on src/main/java/corgi/commands/CommandType.java:
```diff
@@ -0,0 +1,72 @@
+            return DATE;
+        case "find":
+            return FIND;
+        default:
```
Good job on adding a default branch!

ketweeen commented on src/main/java/corgi/parsers/CommandParser.java:
```diff
@@ -0,0 +1,239 @@
+    public Command parse(String fullCommand) throws InvalidCommandFormatException, InvalidCommandTypeException {
+        String[] inputs = fullCommand.split(" ", 2);
+        String cmdStr = inputs[0];
+
```
Nice spacing!

ketweeen commented on src/main/java/corgi/storage/Storage.java:
```diff
@@ -0,0 +1,107 @@
+                            + "\nCorrupted data: " + line + " (Skipped)");
+                }
+            }
+        } catch (IOException e) {
```
No empty catch blocks! Nice!! 😄

ryanozx commented on src/main/java/corgi/Corgi.java:
```diff
@@ -0,0 +1,84 @@
+     * Starts the chatbot - Corgi.
+     */
+    public void start() {
+        Scanner sc = new Scanner(System.in);
```
Would it be better to give the scanner a more descriptive name, given that it has a scope of > 20 lines?

ryanozx commented on src/main/java/corgi/Corgi.java:
```diff
@@ -0,0 +1,84 @@
+     * Starts the chatbot - Corgi.
+     */
+    public void start() {
+        Scanner sc = new Scanner(System.in);
```
Would it be better to give the scanner a more descriptive name, given that it has a scope of > 20 lines?

NereusWB922 replied:
Thank you for the suggestion, will consider it.

ryanozx commented on src/main/java/corgi/commands/AddTaskCommand.java:
```diff
@@ -0,0 +1,56 @@
+import corgi.ui.Ui;
+
+/**
+ * Represents a command to add a task to the task list.
```
Good job! Comments are very descriptive.

ryanozx commented on src/main/java/corgi/commands/FindTasksContainKeywordCommand.java:
```diff
@@ -0,0 +1,55 @@
+/**
+ * Represents a command to find tasks containing a specific keyword in the task list.
+ */
+public class FindTasksContainKeywordCommand extends Command {
```
Class name makes it clear what the class represents, good job!

NereusWB922 committed:
Set main class to Launcher

ryanozx commented on src/main/java/corgi/commands/MarkTaskCommand.java:
```diff
@@ -0,0 +1,63 @@
+    /**
+     * The new status of the task (true for done, false for undone).
+     */
+    private boolean status;
```
Would it be better to rename this variable as "isDone" instead?

NereusWB922 replied:
Thanks for the suggestion, will consider it.

NereusWB922 committed:
Move to using powershell

NereusWB922 committed:
GUI initial setup

NereusWB922 committed:
Add profile pic

NereusWB922 committed:
Add MainWindow fxml and controller

NereusWB922 committed:
Add DialogBox fxml and controller

NereusWB922 committed:
Add Main file

NereusWB922 committed:
Add Launcher class to launch Main

NereusWB922 committed:
Refactor message printing to return as String

NereusWB922 committed:
Change user pic

NereusWB922 committed:
Rename method

NereusWB922 committed:
Use FX Circle to contain pic

NereusWB922 committed:
Edit intro message

NereusWB922 committed:
Add getIntro method

NereusWB922 committed:
Rename field

NereusWB922 committed:
Add GUI styling

NereusWB922 committed:
Merge branch-Level-10 into master

NereusWB922 committed:
Use varargs in returnMessage

NereusWB922 committed:
Merge branch-A-Varargs into master

NereusWB922 committed:
Fix checkstyle error

NereusWB922 committed:
Enable assertions in gradle

NereusWB922 committed:
Add assertions fro Corgi.java

The code lacks explicit validation checks for input and command parsing.

Without propoer validation, the code may encounter unexpected errors when processing user input.

Add assertions for input validation and command parsing.

NereusWB922 committed:
Add assertions for CommandParser.java

Lacks explicit validation checks for creating new object.

Without propoer validation, the code may encounter unexpected errors.

Add assertions for argument validation and new object validation.

NereusWB922 committed:
Add assertions for TaskParser.java

Lacks explicit validation checks for creating new object.

Without propoer validation, the code may encounter unexpected errors.

Add assertions for arguments validation and new object validation.

NereusWB922 committed:
Validate input before requesting response

NereusWB922 committed:
Fix checkstyle error

NereusWB922 committed:
Shorten method name

NereusWB922 committed:
Add comment to explain the predicate

NereusWB922 committed:
Add JavaDoc comment

NereusWB922 committed:
Remove complicated expressions

NereusWB922 committed:
Fix checkstyle error

NereusWB922 committed:
Replace parameter with explicit methods

NereusWB922 committed:
Replace parameter of newAddCommand with methods

NereusWB922 committed:
Remove unused method and field

The Task class contains an unused method named getCommandType.

The getCommandType method serves no purpose.

Removes the unused getCommandType method and command type field from the Task class.

NereusWB922 committed:
Merge pull request #1 from NereusWB922/branch-A-Assertions

NereusWB922 committed:
Merge branch 'master' into branch-A-CodeQuality

NereusWB922 committed:
Merge pull request #2 from NereusWB922/branch-A-CodeQuality

NereusWB922 committed:
Fix checkstyle error

NereusWB922 committed:
Setup CI workflow with GitHub Actions

NereusWB922 committed:
Grant execute permission for gradlew script

NereusWB922 committed:
Merge pull request #3 from NereusWB922/branch-A-CI

NereusWB922 committed:
Add history to track the previous state & command

NereusWB922 committed:
Store Command object in history

NereusWB922 committed:
Pass history stack as argument in execute method

NereusWB922 committed:
Add new undo command

NereusWB922 committed:
Implement UndoCommand class

NereusWB922 committed:
Add new undo command

NereusWB922 committed:
Remove unused import

NereusWB922 committed:
Add string representation of command

NereusWB922 committed:
Add logic to return UNDO command

NereusWB922 committed:
Add missing break statement

NereusWB922 committed:
Store chatbot state in an immutable class

NereusWB922 committed:
Convert to immutable class

NereusWB922 committed:
Refactor execute method to accept immutable State

NereusWB922 committed:
Add JavaDoc comment

NereusWB922 committed:
Fix checkstyle error

NereusWB922 committed:
Merge pull request #4 from NereusWB922:branch-BCD-Extension

Branch-BCD-Extension

NereusWB922 committed:
Fix uninitialized field bug

NereusWB922 committed:
Add more personality to chatbot

NereusWB922 committed:
Fix checkstyle error

NereusWB922 committed:
Merge pull request #6 from NereusWB922/branch-A-Personality

Merge branch-A-Personality to master

NereusWB922 committed:
Exit application for bye command

NereusWB922 committed:
Edit dialog box layout

NereusWB922 committed:
Add application title and icon

NereusWB922 committed:
Make GUI resizeable

NereusWB922 committed:
Fix checkstyle error

NereusWB922 committed:
Merge pull request #7 from NereusWB922/branch-A-BetterGui

Branch-A-BetterGui

NereusWB922 committed:
Edit command format for easier validation

NereusWB922 committed:
Edit getter format

NereusWB922 committed:
Create Validator class for command validation

NereusWB922 committed:
Edit text format

NereusWB922 committed:
Refactor command parsing logic and error handling

NereusWB922 committed:
Check start date is before end date

NereusWB922 committed:
Fix checkstyle error

NereusWB922 committed:
Merge pull request #8 from NereusWB922/branch-A-MoreErrorHandling

Branch-A-MoreErrorHandling

NereusWB922 committed:
Add more test for evey command parsing

NereusWB922 committed:
Fix exception not thrown for invalid num of args

NereusWB922 committed:
Add unit test for validateArguments method

NereusWB922 committed:
Fix checkstyle error

NereusWB922 committed:
Merge pull request #9 from NereusWB922/branch-A-MoreTesting

Branch-A-MoreTesting

NereusWB922 committed:
Add Ui screenshot

NereusWB922 committed:
Add user guide in README

NereusWB922 committed:
Merge pull request #10 from NereusWB922/branch-A-UserGuide

Branch-A-UserGuide

NereusWB922 committed:
Edit return message for list task

NereusWB922 committed:
Edit naming of boolean type variable

NereusWB922 committed:
Remove dead code

NereusWB922 committed:
Merge pull request #11 from NereusWB922/5-sharing-ip-code-quality-feedback-for-nereuswb922

Solve issue #5

NereusWB922 committed:
Update README file
